samples/boards/olimex_stm32_e407/ccm/README.rst

```diff
@@ -0,0 +1,186 @@
+    *
+    * SPDX-License-Identifier: Apache-2.0
+    */
+
+   /dts-v1/;
+   #include <st/stm32f407.dtsi>
+
+   / {
+           model = "Olimex STM32-E407 board";
+           compatible = "olimex,stm32-e407", "st,stm32f407";
+
+           chosen {
+                   zephyr,console = &usart1;
+                   zephyr,sram = &sram0;
+                   zephyr,flash = &flash0;
+                   zephyr,ccm = &ccm0;
+           };
+   };
+
+   &usart1 {
+           current-speed = <115200>;
+           pinctrl-0 = <&usart1_pins_a>;
+           pinctrl-names = "default";
+           status = "ok";
+   };
+
+   &usart3 {
+           current-speed = <115200>;
+           pinctrl-0 = <&usart3_pins_a>;
+           pinctrl-names = "default";
+           status = "ok";
+   };
+
+   &usart6 {
+           current-speed = <115200>;
+           pinctrl-0 = <&usart6_pins_b>;
+           pinctrl-names = "default";
+           status = "ok";
+   };
+
+
+Building
+********
+
+.. code-block:: console
+
+   $ cd samples/boards/olimex_stm32_e407/ccm
+   $ make
+
+Running
+*******
+
+The first time the example is ran after power on the output will look
+someting like this;
+
+.. code-block:: console
+
+   ***** BOOTING ZEPHYR OS v1.11.0 *****
+
+   CCM (Core Coupled Memory) usage example
+
+   The total used CCM area   : [0x10000000, 0x10000021)
+   Zero initialized BSS area : [0x10000000, 0x10000007)
+   Unitialized NOINIT area   : [0x10000008, 0x10000013)
+   Initialised DATA area     : [0x10000014, 0x10000021)
+   Start of DATA in FLASH    : 0x08003940
+
+   Checking initial variable values: ... PASSED
+
+   Initial variable values:
+   ccm_data_var_8  addr: 0x10000014 value: 0x12
+   ccm_data_var_16 addr: 0x10000016 value: 0x3456
+   ccm_data_var_32 addr: 0x10000018 value: 0x789abcde
+   ccm_data_array  addr: 0x1000001c size: 5 value:
+           0x11 0x22 0x33 0x44 0x55
+   ccm_bss_array addr: 0x10000000 size: 7 value:
+           0x00 0x00 0x00 0x00 0x00 0x00 0x00
+   ccm_noinit_array addr: 0x10000008 size: 11 value:
+           0xa9 0x99 0xba 0x90 0xe1 0x2a 0xba 0x93 0x4c 0xfe 0x4b
+
+   Variable values after writing:
+   ccm_data_var_8  addr: 0x10000014 value: 0xed
+   ccm_data_var_16 addr: 0x10000016 value: 0xcba9
+   ccm_data_var_32 addr: 0x10000018 value: 0x87654321
+   ccm_data_array  addr: 0x1000001c size: 5 value:
+           0xaa 0xaa 0xaa 0xaa 0xaa
+   ccm_bss_array addr: 0x10000000 size: 7 value:
+           0xbb 0xbb 0xbb 0xbb 0xbb 0xbb 0xbb
+   ccm_noinit_array addr: 0x10000008 size: 11 value:
+           0xcc 0xcc 0xcc 0xcc 0xcc 0xcc 0xcc 0xcc 0xcc 0xcc 0xcc
+
+   Example end
+
+
+First the CCM sections with their addresse and size are listed, after that
+there are some usage examples. Important to notice is that the noinit section
+holds random data. After writing to the variables they all should hold the
+values shown above.
+
+When the board is reseted without powercycling, the output looks like this;
```

Comment:
Change to:    

     When the board is reset (without power-cycling), the output looks like this: